### PR TITLE
Fix dev ctx with cuda graph

### PR DESCRIPTION
**Pull request by FeixLiu**
### PR types
Others

### PR changes
Others

### Describe
Fix dev ctx with cuda graph.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_cuda_graph.py [hunk 1]
```diff
@@ -236,6 +236,16 @@ def __getitem__(self, idx):
...
 
+    def test_dev_ctx_alloc(self):
+        if not can_use_cuda_graph():
+            return
+
+        x = paddle.to_tensor([2], dtype='float32')
+        graph = CUDAGraph()
+        graph.capture_begin()
+        y = paddle.cast(x, dtype='float16')
+        graph.capture_end()
```
Comment (sneaxiy) [inline python/paddle/fluid/tests/unittests/test_cuda_graph.py:247]: Add more complex cases?

Comment (FeixLiu) [reply python/paddle/fluid/tests/unittests/test_cuda_graph.py:247]: Very few op use dev_ctx.Alloc, cast is one of them. So just call a single cast here.

**Review by sneaxiy — APPROVED** [review]

File: paddle/fluid/platform/device_context_test_cuda_graph.cu [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (chenwhql) [inline paddle/fluid/platform/device_context_test_cuda_graph.cu:1]: 2016 -> 2022

**Review by chenwhql — APPROVED** [review]

File: paddle/fluid/platform/device_context_test_cuda_graph.cu (hunk 2 above)
Comment (FeixLiu) [reply paddle/fluid/platform/device_context_test_cuda_graph.cu:1]: 😂 good point, will fix this in following pr

**Review by jiweibo — APPROVED** [review]

**Pull request merged**